### PR TITLE
fix MPICH implementation

### LINKED ISSUE #1637: Support MPICH when using horovodrun
I'm using MPICH instead of OpenMPI. 
I'm able to train models using mpiexec and mpirun on my cluster but when I try to use horovodrun, it fails with the following error

```python
[ec2-user@master ~]$python3 ./horovodrun -np 8 python3 <train command>
[mpiexec@master.amazon.com] match_arg (utils/args/args.c:159): unrecognized argument allow-run-as-root
[mpiexec@master.amazon.com] HYDU_parse_array (utils/args/args.c:174): argument matching returned error
[mpiexec@master.amazon.com] parse_args (ui/mpich/utils.c:1596): error parsing input array
[mpiexec@master.amazon.com] HYD_uii_mpx_get_parameters (ui/mpich/utils.c:1648): unable to parse user arguments
[mpiexec@master.amazon.com] main (ui/mpich/mpiexec.c:153): error parsing parameters
```

Horovod version: 0.18.2
MPICH version: 3.2

From looking at it, it looks like certain args are not available in MPICH.
Is horovodrun compatible with MPICH?

### PR DESCRIPTION
**Pull request by njzjz**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes #1637.

I notice horovod's MPICH implementation is broken (#1637) but Intel MPI implementation is good (#2374). Per [Intel MPI's official page](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/mpi-library.html), it implements the MPICH specification (in fact it's MPICH derivative), so we can safely use Intel MPI's implementation for MPICH.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Review by chazeon — APPROVED** [review]

File: horovod/runner/mpi_run.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def mpi_run(settings, nics, env, command, stdout=None, stderr=None):
     if mpi_impl_flags is None:
         raise Exception(_MPI_NOT_FOUND_ERROR_MSG)
 
-    impi = _IMPI_IMPL == mpi
+    impi = (_IMPI_IMPL == mpi or _MPICH_IMPL == mpi)
```
Comment (EnricoMi) [inline horovod/runner/mpi_run.py:156]: Can this logic be rephrased to this, which I think is more readable?
```suggestion
    impi = mpi in (_IMPI_IMPL, _MPICH_IMPL)
```

Comment (EnricoMi) [reply horovod/runner/mpi_run.py:156]: I think, `impi` should be renamed to `impi_or_mpich` as this is what it now encodes.

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by EnricoMi** [comment]
Good catch, thanks for fixing this.

File: horovod/runner/mpi_run.py (hunk 1 above)
Comment (njzjz) [reply horovod/runner/mpi_run.py]: Fixed.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**